partiql-plan/src/main/kotlin/org/partiql/plan/v1/operator/rel/RelCorrelate.kt

```diff
@@ -0,0 +1,70 @@
+package org.partiql.plan.v1.operator.rel
+
+import org.partiql.plan.v1.Schema
+
+/**
+ * Logical operator for nested-loop joins (correlated subqueries // lateral joins).
+ */
+public interface RelCorrelate : Rel {
```

Comment:
```suggestion
public interface RelJoinLateral : Rel {
```

I might be missing something. Is this modeling drawn from somewhere else?

Does it make sense to combine correlated subqueries with lateral joins? When we execute a subquery (`ExprSelect`), the environment is passed to it, and we nest the input environment in the current environment. So, the modeling of `RexSelect` would probably act similarly, where we don't need a correlate rel to accomplish that.

Reply:
https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/rel/core/Correlate.java